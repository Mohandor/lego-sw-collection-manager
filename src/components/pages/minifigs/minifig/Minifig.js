import React from "react";
import PropTypes from 'prop-types';
import { Grid, Paper, Typography, Divider } from "@material-ui/core";
import LogoLink from './logo-link';
import NameAndTags from './name-and-tags';
import styles from './Minifig.module.css';
import Management from "./management";

export const minifigs = props => {
  const { reference, details } = props;
  return (
    <Paper className={styles.paper}>
      <Grid container className={styles.container} direction="column" justify="space-between">
        <Grid item className={styles.pictureContainer}>
          <img
            className={styles.picture}
            src={`https://img.bricklink.com/ItemImage/MN/0/${reference}.png`}
            alt={`${reference}-bricklink-png`}
          />
        </Grid>
        <Grid item className="full-width">
          <Typography>{reference.toUpperCase()}</Typography>
          <NameAndTags characterName={details.characterName} tags={details.tags} />
          <LogoLink reference={reference} type={'bricklink'} />
          <LogoLink reference={reference} type={'brickset'} />
          <Divider variant="fullWidth" />
        </Grid>
        <Grid item className="full-width">
          <Management reference={reference} details={details} />
        </Grid>
      </Grid>
    </Paper>
  )
}

minifigs.propTypes = {
  reference: PropTypes.string.isRequired,
  details: PropTypes.shape({
    characterName: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    possessed: PropTypes.bool.isRequired,
    tags: PropTypes.arrayOf(PropTypes.string)
  })
}

export default minifigs;
