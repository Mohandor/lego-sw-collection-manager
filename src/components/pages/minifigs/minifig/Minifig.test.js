import React from 'react';
import Minifig from './Minifig';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<Minifig />', () => {
  it('should render a minifig', () => {
    const props = {
      reference: 'sw0001',
      details: {
        characterName: 'Battle Droid',
        tags: ['test'],
        name: 'test',
        possessed: true
      }
    }
    const wrapper = shallow(<Minifig {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});