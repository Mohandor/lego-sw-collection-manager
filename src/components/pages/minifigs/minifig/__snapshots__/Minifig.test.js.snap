// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Minifig /> should render a minifig 1`] = `
<WithStyles(ForwardRef(Paper))
  className="paper"
>
  <WithStyles(ForwardRef(Grid))
    className="container"
    container={true}
    direction="column"
    justify="space-between"
  >
    <WithStyles(ForwardRef(Grid))
      className="pictureContainer"
      item={true}
    >
      <img
        alt="sw0001-bricklink-png"
        className="picture"
        src="https://img.bricklink.com/ItemImage/MN/0/sw0001.png"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="full-width"
      item={true}
    >
      <WithStyles(ForwardRef(Typography))>
        SW0001
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(undefined)
        characterName="Battle Droid"
        tags={
          Array [
            "test",
          ]
        }
      />
      <logoLink
        reference="sw0001"
        type="bricklink"
      />
      <logoLink
        reference="sw0001"
        type="brickset"
      />
      <WithStyles(ForwardRef(Divider))
        variant="fullWidth"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="full-width"
      item={true}
    >
      <Connect(Management)
        details={
          Object {
            "characterName": "Battle Droid",
            "name": "test",
            "possessed": true,
            "tags": Array [
              "test",
            ],
          }
        }
        reference="sw0001"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Paper))>
`;
